docs/docs/aztec/concepts/pxe/index.md

```diff
@@ -8,44 +8,101 @@ importance: 1
 
-The PXE generates proofs of private function execution, and sends these proofs along with public function requests to the sequencer. Private inputs never leave the client-side PXE.
+The PXE is a client-side interface of the PXE Service. The PXE generates proofs of private function execution, and sends these proofs along with public function execution requests to the sequencer. Private inputs never leave the client-side PXE.
 
-<Image img={require("/img/pxe.png")} />
+The PXE is responsible for:
 
-## PXE Service
+- storing secrets (e.g. encryption keys, notes, tagging secrets for note discovery) and exposing an interface for safely accessing them
+- orchestrating private function (circuit) execution and proof generation, including implementing [oracles](../../smart_contracts/oracles/index.md) needed for transaction execution
+- syncing users' relevant network state, obtained from an Aztec node
+- safely handling multiple accounts with siloed data and permissions
 
-The PXE is a client-side interface of the PXE Service, which is a set of server-side APIs for interacting with the network. It provides functions for account management, contract and transaction interactions, note management, and more.
+One PXE can handle data and secrets for multiple accounts.
+
+## System architecture
+
+```mermaid
+flowchart TB
+    User --Interacts with--> Wallet
+    Wallet --Prompts--> User
+    subgraph Browser
+    Dapp
+    end
+    Dapp --Calls (requires auth)--> Wallet
+    subgraph Client-side
+        Wallet --Scoped--> PXE
+        PXE --Execute/Prove--> Circuits
+        Circuits --Oracle--> PXE
+    end
+    PXE --Queries world-state (causes privacy leaks)--> Node
+    Wallet --Track tx state (may be handled via PXE)--> Node
+
+```
 
 ## Components
 
-### ACIR simulator
+### Transaction Simulator
+
+An application will prompt the users PXE to execute a transaction (e.g. execute X function, with Y arguments, from Z account). The application or the wallet may handle gas estimation.
+
+The ACIR (Abstract Circuit Intermediate Representation) simulator handles the execution of smart contract functions by simulating transactions. It generates the required data and inputs for these functions. You can find more details about how it works [here](./acir_simulator.md).
 
-The ACIR (Abstract Circuit Intermediate Representation) simulator handles the accurate execution of smart contract functions by simulating transactions. It generates the required data and inputs for these functions. You can find more details about how it works [here](./acir_simulator.md).
+### Proof Generation
+
+After simulation, the wallet calls `proveTx` on the PXE with all of the data generated during simulation and any [authentication witnesses](../accounts/authwit.md) (for allowing contracts to act on behalf of the users' account contract).
```

Comment:
Until we do simulated simulations (https://github.com/AztecProtocol/aztec-packages/issues/9133), we'll also need authwits for simulation, before attempting to prove.